airflow/providers/amazon/aws/triggers/README.md

```diff
@@ -0,0 +1,153 @@
+
+1. `trigger`: This is the trigger which you want to pass the execution to. We will write this trigger in just a moment.
+2. `method_name`: This specifies the name of the method you want to execute once the trigger completes its execution. The trigger cannot pass the execution back to the execute method of the operator. By convention, the name for this method is `execute_complete`.
+3. `timeout`: An optional parameter that controls the length of time the Trigger can execute for before timing out. This defaults to `None`, meaning no timeout.
+4. `kwargs`: Additional keyword arguments to pass to `method_name`. Default is `{}`.
+
+The Trigger is the main component of deferrable operators. They must be placed in the `airflow/providers/amazon/aws/triggers/` folder. All Triggers must implement the following 3 methods:
+
+1. `__init__`: the constructor which receives parameters from the operator. These must be JSON serializable.
+2. `serialize`: a function that returns the classpath, as well as keyword arguments to the `__init__`  method as a tuple
+3. `run` : the asynchronous function that is responsible for awaiting the asynchronous operations.
+
+Ideally, when the operator has deferred itself, it has already initiated the "main" task of the operator, and is now waiting for a certain resource to reach a certain state.
+As mentioned earlier, the botocore library defines a `Waiter` class for many services, which implements a `wait` method that can be configured via a config file to poll the boto3 API at set intervals, and return if the success criteria is met.
+The aiobotocore library, which is used to make asynchronous botocore calls, defines an `AIOWaiter` class, which also implements a wait method that behaves identical to the botocore method, except that it works asynchronously.
+Therefore, any botocore waiter is available as an aiobotocore waiter, and can be used to asynchronously poll a service until the desired criteria is met.
+
+To call the asynchronous `wait` function, first create a hook for the particular service. For example, for a Redshift hook, it would look like this:
+
+```python
+self.redshift_hook = RedshiftHook(aws_conn_id=self.aws_conn_id)
+```
+
+With this hook, we can use the async_conn property to get access to the aiobotocore client:
+
+```python
+async with self.redshift_hook.async_conn as client:
+    await client.get_waiter("cluster_available").wait(
+        ClusterIdentifier=self.cluster_identifier,
+        WaiterConfig={
+            "Delay": int(self.poll_interval),
+            "MaxAttempts": int(self.max_attempt),
+        },
+    )
+```
+
+In this case, we are using the built-in cluster_available waiter. If we wanted to use a custom waiter, we would change the code slightly to use the get_waiter function from the hook, rather than the aiobotocore client:
```

Comment:
```suggestion
In this case, we are using the built-in cluster_available waiter. If we wanted to use a custom waiter, we would change the code slightly to use the `get_waiter` function from the hook, rather than the aiobotocore client:
```